src/test/kotlin/com/exactpro/th2/check1/rule/check/TestCheckRuleTask.kt

```diff
@@ -189,4 +203,61 @@ internal class TestCheckRuleTask : AbstractCheckTaskTest() {
             "No failed event $eventBatch"
         }
     }
+
+    @ParameterizedTest(name = "keepRepeatingGroupResult = {0}")
+    @MethodSource("resultByFilter")
+    fun `check that the order is kept in repeating groups`(resultByFilter: Pair<EventStatus, Map<String, ValueFilter>>) {
+        val streams = createStreams(SESSION_ALIAS, Direction.FIRST, listOf(
+                message(MESSAGE_TYPE, Direction.FIRST, SESSION_ALIAS)
+                        .putFields("legs", listValue()
+                                .add(message(MESSAGE_TYPE, Direction.FIRST, SESSION_ALIAS)
+                                        .putAllFields(mapOf(
+                                                "A" to "1".toValue(), "B" to "2".toValue()
+                                        )))
+                                .add(message(MESSAGE_TYPE, Direction.FIRST, SESSION_ALIAS)
+                                        .putAllFields(mapOf(
+                                                "C" to "3".toValue(), "D" to "4".toValue()
+                                        )))
+                                .toValue())
+                        .build()
+        ))
+        val messageFilterForCheckOrder: RootMessageFilter = RootMessageFilter.newBuilder()
+                .setComparisonSettings(RootComparisonSettings.newBuilder().build())
+                .setMessageType(MESSAGE_TYPE)
+                .setMessageFilter(MessageFilter.newBuilder()
+                        .putFields("legs", ValueFilter.newBuilder()
+                                .setListFilter(ListValueFilter.newBuilder().apply {
+                                    addValues(ValueFilter.newBuilder()
+                                            .setMessageFilter(MessageFilter.newBuilder()
+                                                    .putAllFields(resultByFilter.second).build())
+                                            .build())
+                                }).build())
+                        .build())
+                .build()
+        val eventID = EventID.newBuilder().setId("root").build()
+
+        checkTask(messageFilterForCheckOrder, eventID, streams).begin()
+
+        val eventBatches = awaitEventBatchRequest(1000L, 2)
+        val eventList = eventBatches.flatMap(EventBatch::getEventsList)
+        assertAll({
+            assertEquals(3, eventList.size)
+        }, {
+            val verificationEvents = eventList.filter { it.type == "Verification" }
+            assertEquals(1, verificationEvents.size)
```

Comment:
How do we check the order of repeating groups if we submit only one filter?
checke1 sorts in order to increase the number of FAILED fields in the previous version.
Let's make a message with two group filters:
* both fields are failed
* the only one filed failed
and run it vs two messages which have (A, B) and (B, A) order of the repeating group.
The order of results will be equivalent the order in message